README.md

```diff
@@ -10,6 +17,92 @@ To build this snap locally you need snapcraft. The project must be built as real
 $ sudo snapcraft
 ```
 
+# Testing with spread
+
+## Prerequisites for testing
+
+You need to have the following software installed before you can test with spread
+ - Go (https://golang.org/doc/install or ```sudo snap install go```)
+ - Spread (install from source as per below)
+
+## Installing spread
+
+Manually installing spread is the recommended way to get started as
+we need some LXD profile support that has not reached upstream. This document will be updated with the upstream version when this happens.
+
+```
+git clone https://github.com/Meulengracht/spread
+cd spread
+go mod init
+cd cmd/spread
+go build .
+go install .
+```
+
+## QEmu backend
+
+1. Install the dependencies required for the qemu emulation
+```
+sudo apt update && sudo apt install -y qemu-kvm autopkgtest
+```
+2. Create a suitable ubuntu test image (focal) in the following directory where spread locates images
+```
+mkdir -p ~/.spread/qemu
+cd ~/.spread/qemu
+autopkgtest-buildvm-ubuntu-cloud -r focal
+```
+3. Rename the newly built image as the name will not match what spread is expecting
+```
+mv autopkgtest-focal-amd64.img ubuntu-20.04-64.img
+```
+4. Now you are ready to run spread tests with the qemu backend
+```
+cd ~/core20 # or wherever you checked out this repository
+spread qemu-nested
+```
+
+## LXD backend
+The LXD backend is the preffered way of testing locally as it uses virtualization and thus runs a lot quicker than
+the qemu backend. This is because the container can use all the resources of the host, and we can support
+qemu-kvm acceleration in the container for the nested instance.
+
+This backend requires that your host machine supports KVM.
+
+1. Install LXD snap from the stable channel (other channels probably work but are untested)
+```
+sudo snap install lxd --channel=latest/stable
+```
+2. [Download](https://drive.google.com/drive/folders/1wk3rA-Sw3BB_EIlkc1pPC3MJKxbxba-U?usp=sharing) the latest image and import it into lxd as it is required for testing core repositories if you haven't already got this image.
```

Comment:
Yes - however there are two images in this case.
1. The LXD container image which runs the tests (shared in the gdrive link). 
2. The UC20 image that is built inside the LXD container using ```ubuntu-image``` 

So this is actually one of the remaining issues which I have completely forgot to point out. We need to share this LXD image in an LXD server somewhere where we can pull the image from through appropriate protocols instead of manually importing some image I have built on a personal gdrive.

I am however unsure who to contact for this. @mvo5 do you have any idea how to proceed with this?

Reply:
@Meulengracht how did you build the lxd image? maybe for now we could just dynamically build the image as part of the CI flow and then import it into lxd as part of the test itself?

Reply:
@anonymouse64 I've built the image manually, but currently trying to automate the process and put it somewhere. It's not really any magic, it's just a focal base image with (lots of) dependencies installed to speedup the process of running tests. 

I'm not quite sure what you mean, I mean its definitaly a good idea to automate the process of the lxd image so you can just [re]build it locally (or as a part of CI, but we are not using LXD in the CI) - but I'm not sure if its neccessary to build the image everytime.

Reply:
I have now removed the link to my gdrive and added an automated script to build and setup the LXD image.

Reply:
Yeah ideally we would cache the image somewhere so it's easily usable instead of duplicating all the time, but until such a time, best to have the steps to create such an image somewhere easy. 

@sergiocazzolato I know you help us build other images and store the artifacts in GCE buckets, perhaps you could help @Meulengracht integrate this specific image and steps to keep the image up to date into your machinery where you do the same for snapd things ?